presto-main/src/main/java/io/prestosql/sql/planner/optimizations/UnaliasSymbolReferences.java

```diff
@@ -521,8 +521,9 @@ public PlanNode visitJoin(JoinNode node, RewriteContext<Void> context)
 
             if (node.getType() == INNER) {
                 canonicalCriteria.stream()
-                        .filter(clause -> types.get(clause.getLeft()).equals(types.get(clause.getRight())))
-                        .filter(clause -> node.getOutputSymbols().contains(clause.getLeft()))
+                        // Map right equi-condition symbols to left symbols. This helps to
+                        // reuse join node partitioning better as partitioning properties are
+                        // only derived from probe side symbols
```

Comment:
I think this is correct except for RIGHT joins. Maybe map equi-condition symbols to left symbol for all joins except RIGHT, and map to right symbols otherwise?

Reply:
good catch!

Reply:
Actually, this is invalid unalias, as one might explicitly check outer equi symbol